packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -640,8 +640,11 @@ class RenderParagraph extends RenderBox
 
     RenderBox child = firstChild;
     int childIndex = 0;
-    while (child != null) {
-      assert(childIndex < _textPainter.inlinePlaceholderBoxes.length);
+    // childIndex might be out of index of placeholder boxes. This can happen
+    // if engine truncates children due to ellipsis. Sadly, we would not know
+    // it until we finish layout, and RenderObject is in immutable state at
+    // this point.
```

Comment:
I'm not really sure what "and RenderObject is in immutable state at this point" means here.

Reply:
The ideal solution here will be removing RenderObject from the children, but we can't since the renderobject is locked after layout phase